src/colmap/estimators/generalized_relative_pose.cc

```diff
@@ -37,8 +37,64 @@
 #include "colmap/util/logging.h"
 
 #include <Eigen/Dense>
+#include <PoseLib/solvers/gen_relpose_6pt.h>
 
 namespace colmap {
+
+void GR6PEstimator::Estimate(const std::vector<X_t>& points1,
+                             const std::vector<Y_t>& points2,
+                             std::vector<M_t>* rigs2_from_rigs1) {
+  THROW_CHECK_EQ(points1.size(), 6);
+  THROW_CHECK_EQ(points2.size(), 6);
+  THROW_CHECK_NOTNULL(rigs2_from_rigs1);
+
+  rigs2_from_rigs1->clear();
+
+  std::vector<Eigen::Vector3d> proj_centers1(6);
+  std::vector<Eigen::Vector3d> proj_centers2(6);
+  std::vector<Eigen::Vector3d> rays1(6);
+  std::vector<Eigen::Vector3d> rays2(6);
+  for (int i = 0; i < 6; ++i) {
+    proj_centers1[i] = points1[i].cam_from_rig.rotation.inverse() *
+                       -points1[i].cam_from_rig.translation;
+    proj_centers2[i] = points2[i].cam_from_rig.rotation.inverse() *
+                       -points2[i].cam_from_rig.translation;
+    rays1[i] =
+        points1[i].cam_from_rig.rotation.inverse() * points1[i].ray_in_cam;
+    rays2[i] =
+        points2[i].cam_from_rig.rotation.inverse() * points2[i].ray_in_cam;
+  }
+
+  std::vector<poselib::CameraPose> poses;
+  poselib::gen_relpose_6pt(proj_centers1, rays1, proj_centers2, rays2, &poses);
+
+  rigs2_from_rigs1->reserve(poses.size());
+  for (const poselib::CameraPose& pose : poses) {
+    rigs2_from_rigs1->emplace_back(
+        Eigen::Quaterniond(pose.q(0), pose.q(1), pose.q(2), pose.q(3)), pose.t);
```

Comment:
@ahojnnes, If I recall correctly the poselib quaternion and Eigen::Quaterniond is not in the same order? Maybe double check this

Reply:
This reminds me though: we don't have any tests for the minimal solver being correct, and LO can oftentimes lead to correct results even with a wrongly configured minimal solver, if the convergence is good enough. Should we add some tests in the future for the Estimate() function. This can catch such inconsistent convention issues and also be potentially beneficial for PoseLib, as no such tests are available in PoseLib either.